pandas/core/dtypes/cast.py

```diff
@@ -118,7 +131,7 @@ def is_nested_object(obj) -> bool:
     return False
 
 
-def maybe_downcast_to_dtype(result, dtype):
+def maybe_downcast_to_dtype(result, dtype: Dtype):
```

Comment:
should this be DtypeObj?

Reply:
has to be `Dtype` I think because could be a `str`